template-sample-app/template/src/components/layers/StoresLayer.js

```diff
@@ -20,19 +18,22 @@ export default function StoresLayer() {
   const navigate = useNavigate();
   const { storesLayer } = useSelector((state) => state.carto.layers);
   const source = useSelector((state) => selectSourceById(state, storesLayer?.source));
+  const DEFAULT_PROPS = useCartoProps(source);
 
   if (storesLayer && source) {
     return new CartoSQLLayer({
-      id: 'storesPointLayer',
-      data: buildQueryFilters(source),
+      ...DEFAULT_PROPS,
+      id: storesLayer.id,
+      data: source.data,
       credentials: source.credentials,
+      uniqueIdProperty: 'cartodb_id',
```

Comment:
This should be a CONST as a first step shared for the whole template project, and as a second one, it should probably be exported from the lib

Reply:
For BQ layers, id field could be `geoid` right?

Reply:
Not sure, we should ask bq people on it